File: read_strings.rb
#!/usr/local/bin/ruby -w

# get filename from arg from commandline
filename = ARGV[0]

file_str = ''

# read line-by-line
File.readlines(filename).each do |l|
  # l = current line
  # add l to final output string
  file_str += l
  # print l
  puts(l)
end

# write to output file
File.write(filename + '.out', file_str)

File: read_strings.py
import sys

filename = sys.argv[1]

with open(filename) as f:
    content = f.readlines()
# you may also want to remove whitespace characters like `\n` at the end of each line
content = [x.strip() for x in content]

outfile_name = filename + '.out'
outfile = open(outfile_name,'w')
for c in content:
    print(c)
    outfile.write(c + '\n')

outfile.close()
